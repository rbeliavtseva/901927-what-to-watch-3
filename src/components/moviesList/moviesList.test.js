import React from "react";
import renderer from "react-test-renderer";
import MoviesList from "./moviesList.jsx";

// Mocks
const films = [
  {
    title: `Fight Club`,
    poster: `http://picsum.photos/300/150?r=${Math.random()}`,
    id: 19
  },
  {
    title: `Pulp Fiction`,
    poster: `http://picsum.photos/300/150?r=${Math.random()}`,
    id: 20
  },
  {
    title: `The Godfather`,
    poster: `http://picsum.photos/300/150?r=${Math.random()}`,
    id: 21
  },
  {
    title: `Back to the Future`,
    poster: `http://picsum.photos/300/150?r=${Math.random()}`,
    id: 22
  },
  {
    title: `The Pianist`,
    poster: `http://picsum.photos/300/150?r=${Math.random()}`,
    id: 23
  }
];

it(`Render MoviesList`, () => {
  const tree = renderer
    .create(<MoviesList
      films={films}
      onMovieCardHover={()=>{}}
      onMovieCardTitleClick={()=>{}}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
