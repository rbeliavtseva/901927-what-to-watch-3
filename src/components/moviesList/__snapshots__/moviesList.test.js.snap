// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MoviesList 1`] = `
<div>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Fight Club"
        height="175"
        src="http://picsum.photos/300/150?r=0.9158237574584083"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Fight Club
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Pulp Fiction"
        height="175"
        src="http://picsum.photos/300/150?r=0.036181068699629115"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Pulp Fiction
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="The Godfather"
        height="175"
        src="http://picsum.photos/300/150?r=0.38834366698846656"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        The Godfather
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Back to the Future"
        height="175"
        src="http://picsum.photos/300/150?r=0.26415186022516113"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Back to the Future
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="The Pianist"
        height="175"
        src="http://picsum.photos/300/150?r=0.3484221206525209"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        The Pianist
      </a>
    </h3>
  </article>
</div>
`;
